paramak/parametric_reactors/ball_reactor.py

```diff
@@ -152,6 +166,39 @@ def __init__(
             self.plasma_gap_vertical_thickness,
             self.major_radius - self.minor_radius]
 
+        self.port_type = port_type
+        self.number_of_ports = number_of_ports
+        self.port_center_point = port_center_point
+        self.port_radius = port_radius
+        self.port_height = port_height
+        self.port_width = port_width
+        self.port_distance = port_distance
+        self.port_start_radius = port_start_radius
+        self.port_azimuth_placement_angle = port_azimuth_placement_angle
+        self.port_fillet_radius = port_fillet_radius
+
+        if self.number_of_ports is not None:
+            if self.port_azimuth_placement_angle is None:
+                self.port_azimuth_placement_angle = np.linspace(
+                    0, 360, self.number_of_ports, endpoint=False
+                )
+            else:
+                if self.number_of_ports == len(
+                        self.port_azimuth_placement_angle):
+                    self.port_azimuth_placement_angle = port_azimuth_placement_angle
+                else:
+                    raise ValueError(
+                        'number of ports does not equal number of port azimuthal placement angles')
+        else:
+            if self.port_azimuth_placement_angle is not None:
+                self.number_of_ports = len(self.port_azimuth_placement_angle)
+
+        self.port_start_radius = port_start_radius
+
+        if self.port_azimuth_placement_angle is not None:
+            if self.port_start_radius is None:
+                self.port_start_radius = self.major_radius
```

Comment:
Could all of that be in setters instead of in the init ?

Reply:
The number_of_ports parameter has been removed, removing the need for most of this code. A setter for port_start_radius has been added.